nixos/modules/services/search/meilisearch.nix

```diff
@@ -182,21 +182,7 @@ in
       no_analytics = lib.mkDefault true;
     };
 
-    warnings = lib.optional (lib.versionOlder cfg.package.version "1.12") ''
-      Meilisearch 1.11 will be removed in NixOS 25.11. As it was the last
-      version not to support dumpless upgrades, you will have to manually
-      migrate your data before that. Instructions can be found at
-      https://www.meilisearch.com/docs/learn/update_and_migration/updating#using-a-dump
-      and afterwards, you can set `services.meilisearch.package = pkgs.meilisearch;`
-      to use the latest version.
-    '';
-
-    services.meilisearch.package = lib.mkDefault (
-      if lib.versionAtLeast config.system.stateVersion "25.05" then
-        pkgs.meilisearch
-      else
-        pkgs.meilisearch_1_11
-    );
+    services.meilisearch.package = lib.mkDefault pkgs.meilisearch;
```

Comment:
Bit late to the party with this review, but this line is redundant and can be removed (the option already defaults to `pkgs.meilisearch`). I'll remove it in a new PR.